### PR TITLE
chore(e2e): general e2e test improvements

### PR DESCRIPTION
**Pull request by johncowen**
This began as a clean up of e2e test code following improvements in:

- https://github.com/kumahq/kuma-gui/pull/1482
- https://github.com/kumahq/kuma-gui/pull/1417
- https://github.com/kumahq/kuma-gui/pull/1452

I also noticed some areas mainly around subscription counts that felt a little strange, and began to tweak those a little, and then as I delved deeper I started to go down a bit of a rabbit hole:

- I started to make all mock instances of subscriptions used the `KUMA_SUBSCRIPTION_COUNT` variable.
- Started making all mock instances of subscriptions as consistent as possible, and decided to make a `fake.kuma.connection` for providing `connectTime`/`disconnectTime`, which I _think_ is more realistic, would be good for a second pair of eyes on that. The intention is that only the last subscription possibly won't have a disconnectTime and it will always be after the connectTime. I just padded out the difference in golang date formats with a random number.
- The above led me to looking at our types around subscriptions and then zone overview type things, and this is where I felt like I was going too far into a rabbit hole for this PR. For example, one of the type corrections I made (`zoneEgress` here https://github.com/kumahq/kuma-gui/pull/1558/commits/14dd7105ecc328ae0c0a9635074f5e8c845854c2 seems to be possibly `undefined`) started breaking things deeper in the app, so if this type is incorrect there are more potential undefined access bugs to be had. Instead of fixing things here I undid that change (https://github.com/kumahq/kuma-gui/pull/1558/commits/db2cb1f7a3def1359ab2bf15f5697fb9c78f9abc) for a future piece of work.

### CODE REVIEW

**Comment by kleinfreund** [comment]
**Comment**: I’m almost certain that the change in https://github.com/kumahq/kuma-gui/commit/14dd7105ecc328ae0c0a9635074f5e8c845854c2 would be incorrect anyway. I don’t think `zoneEgress` or `zoneEgressInsight` can actually be absent. Would be good to verify that, but this seems to be a pretty stable pattern: it seems that an object that has properties that themselves refer to objects are never omitted. Another example of this I can think of is `dataplanesByType` on `MeshInsight`. It can never be omitted and neither can its properties.

I don’t however know how general this rule is. It possibly has something to do with our use of protobuf? Don’t know.

File: src/test-support/FakeKuma.ts
Comment (kleinfreund) [inline src/test-support/FakeKuma.ts:71]: **Suggestion**: Development remnant
```suggestion
```

File: src/test-support/FakeKuma.ts
Comment (kleinfreund) [inline src/test-support/FakeKuma.ts:78]: **Suggestion**: Development remnant
```suggestion
```

File: src/types/index.d.ts
Comment (kleinfreund) [inline src/types/index.d.ts:567]: **Issue**: I don’t thin k`zoneEgressInsight` can be omitted, but `subscriptions` can.

```suggestion
  zoneEgressInsight: {
    subscriptions?: DiscoverySubscription[]
  }
```

**Review by kleinfreund — CHANGES_REQUESTED** [review]

File: src/test-support/FakeKuma.ts
Comment (johncowen) [reply src/test-support/FakeKuma.ts:71]: I left these in on purpose so you could see the format and understand what all the fake number padding was about.

File: src/test-support/FakeKuma.ts
Comment (johncowen) [reply src/test-support/FakeKuma.ts:78]: See above

File: src/types/index.d.ts
Comment (johncowen) [reply src/types/index.d.ts:567]: Well I'm not totally sure these are the generated types:

Screenshot 2023-09-28 at 10 08 38
Screenshot 2023-09-28 at 10 10 19

?

Comment (johncowen) [reply src/types/index.d.ts:567]: but then again this is DataplaneInsights, which we know, I guess without the extra protections that the generated types give, can have `subscriptions` as `undefined`

Screenshot 2023-09-28 at 10 14 52

This is were I felt I was going down a large rabbit hole, so I'm more inlined to remove this change entirely. Hopefully we'll be sorting this whole issue out once and for all soonish.

Give me a bit let me take a better look at this from here. brb

**Review by kleinfreund — APPROVED** [review]

File: src/types/index.d.ts
Comment (johncowen) [reply src/types/index.d.ts:567]: Well the point is, we shouldn't have to keep that in mind as these generated types make sure that subscriptions is always an array, which is what we want.

Screenshot 2023-09-28 at 13 18 41

>  It definitely looks there is a knowledge gap somewhere we because we do indeed know subscriptions can be undefined.

Well, I'm not sure if its a "knowledge gap" thing. With those generated types `subscriptions` is always an array, which then prevents all the undefined access errors we have without having to keep that in mind or having to have the knowledge about it. I think that's part of the point.

1 sec I think I'm going to take these type changes out as I'm guessing they could have a bigger impact that I don't want to have to look at in this PR and this was part of the rabbit hole I got the feeling I was getting into here.

Comment (johncowen) [reply src/types/index.d.ts]: OK I removed the type changes here, I have an approval already so I'll wait a bit before merging just incase.

**Comment by johncowen** [comment]
> Comment: I’m almost certain that the change in https://github.com/kumahq/kuma-gui/commit/14dd7105ecc328ae0c0a9635074f5e8c845854c2 would be incorrect anyway. I don’t think zoneEgress or zoneEgressInsight can actually be absent. Would be good to verify that, but this seems to be a pretty stable pattern: it seems that an object that has properties that themselves refer to objects are never omitted. Another example of this I can think of is dataplanesByType on MeshInsight. It can never be omitted and neither can its properties.

> I don’t however know how general this rule is. It possibly has something to do with our use of protobuf? Don’t know.

I think I missed this comment yesterday and I've just seen it now, just for note cos I've already mentioned it in the other inline, but I removed the type changes here anyway - it was those that made me feel like I was getting started into the rabbit hole

**Pull request merged**
